apps/envited-x-data-space/modules/AddAssets/AddAssets.utils.ts

```diff
@@ -28,33 +28,57 @@ export const processFile = async (file: File): Promise<{ name: string; type: str
   return { name: file.name, type: file.type, cid }
 }
 
-export const uploadFile = async (filesArray: File[], { signedUrl, cid, fileType, file }: UploadAssetFile) => {
-  if (!signedUrl) {
-    return { success: false, file, message: ERRORS.SIGNED_URL_MISSING }
-  }
-
-  if (!cid) {
-    return { success: false, file, message: ERRORS.CID_MISSING }
-  }
-
-  if (!fileType) {
-    return { success: false, file, message: ERRORS.FILE_TYPE_MISSING }
-  }
-
-  const fileObj = filesArray.find(propEq(file, 'name'))
-  if (!fileObj) return { success: false, file, message: ERRORS.FILE_NOT_FOUND }
-
-  const arrayBuffer = Buffer.from(await fileObj.arrayBuffer())
-
-  const uploadResponse = await fetch(signedUrl, {
-    method: 'PUT',
-    body: arrayBuffer,
-    headers: { 'Content-Type': fileType },
+export const uploadFile = (
+  filesArray: File[],
+  { signedUrl, cid, fileType, file }: UploadAssetFile,
+  onProgress?: (percent: number) => void,
+): Promise<{ success: boolean; file: string; message?: string }> => {
+  return new Promise(async (resolve, reject) => {
+    if (!signedUrl) {
+      return resolve({ success: false, file, message: ERRORS.SIGNED_URL_MISSING })
+    }
+
+    if (!cid) {
+      return resolve({ success: false, file, message: ERRORS.CID_MISSING })
+    }
+
+    if (!fileType) {
+      return resolve({ success: false, file, message: ERRORS.FILE_TYPE_MISSING })
+    }
+
+    const fileObj = filesArray.find(propEq(file, 'name'))
+    if (!fileObj) {
+      return resolve({ success: false, file, message: ERRORS.FILE_NOT_FOUND })
+    }
+
+    const xhr = new XMLHttpRequest()
```

Comment:
Same here, it's probably a good idea to have an abstract http module that handles http requests so we dont have a fetch here, an xmlhttprequest there etc etc.

Reply:
Now that we have both fetch and xmlhttp, it might be worth considering replacing it all with axios